### ISSUE TITLE
List of actions to be excluded from tracing

### ISSUE THREAD

**Issue by ErmakovSergey** [moleculerjs/moleculer#720]
Can you add exclude feature for **tracing** similar to **metrics**?

**Comment by intech** [comment]
+1

**Comment by synologic** [comment]
Fairly simple i believe,

in tracer.js, change spanFinished function:

```
spanFinished(span) {
                if (span.sampled) {
                        if (this.opts.exclude && this.opts.exclude.length != 0 && this.opts.exclude.indexOf(span.tags.action.name) != -1) return;
                        this.invokeExporter("spanFinished", [span]);
                }
        }
```
then in the tracing config, add an array 

`exclude : ["action1","action2"]`

This will apply for all tracers, same logic can be applied individually.

Alternatively/additionally one can filter by span.name as well (eg this would apply to ApiGw to filter by REST action as span.tags.action.name in ApiGw is api.rest)

**Comment by AndreMaz** [comment]
How this feature would work?
Imagine the following scenario:
![image](https://user-images.githubusercontent.com/9802754/187738222-eb667820-2e1e-464c-9290-f4c5e4450676.png)
and `excludes` field with [`"math.**"`]

In this case the `excludes` means that if we call `math.random` directly there won't by any tracing? 
Or `excludes` means that if we call `greeter.hello` everything after `math.random`, including `math.random`, is not part of the span? Like so:
![image](https://user-images.githubusercontent.com/9802754/187739116-b69e3670-d005-4025-8a61-d55128011b7d.png)

**Comment by intech** [comment]
@AndreMaz I think it should only be for `math.*`
![tracing](https://user-images.githubusercontent.com/451306/187750231-838466a2-28c5-462b-817c-8e3dca97280b.jpg)

**Comment by AndreMaz** [comment]
So, under the hood, it is still necessary to collect all spans and only when pushing or printing them the `mat.**` is excluded?

**Comment by intech** [comment]
Yes, we have discussed this before in discord. In the current implementation, it is not possible. Even if tracing is disabled, it will still collect spans and won't send them.
It is now an acceptable solution to get the functionality to look for unnecessary services from the trace.
